pkg/controller/versions/strategy_v2_5.go

```diff
@@ -239,6 +243,32 @@ func (v *versionStrategyV2_5) validateExtensionProviders(spec *v2.ControlPlaneSp
 	return allErrors
 }
 
+func (v *versionStrategyV2_5) validateExtensionProviderZipkin(spec *v2.ControlPlaneSpec, allErrors []error) []error {
+	if spec.MeshConfig == nil || spec.MeshConfig.ExtensionProviders == nil {
+		return allErrors
+	}
+	// we don't allow to set both existing Jaeger/Stackdriver and extensionProvider.zipkin at the same time
+	if spec.Addons.Jaeger != nil || spec.Addons.Stackdriver != nil {
+		for _, ext := range spec.MeshConfig.ExtensionProviders {
+			if ext.Zipkin != nil {
+				allErrors = append(allErrors, fmt.Errorf("extension provider '%s' can not define zipkin provider "+
+					"when spec.addons.jaeger or spec.addons.stackdriver is defined", ext.Name))
+			}
+		}
+	}
+	// we don't allow to set telemetry v2 and extensionProvider.zipkin at the same time
+	if spec.Telemetry.Type == "Istiod" {
+		for _, ext := range spec.MeshConfig.ExtensionProviders {
+			if ext.Zipkin != nil {
+				allErrors = append(allErrors, fmt.Errorf("extension provider '%s' can not define zipkin provider. "+
+					"Need to disable the telemetry v2 by setting spec.Telemetry.Type: None", ext.Name))
+			}
+		}
+	}
```

Comment:
I wasn't right that we need this validation, because it turns out that `telemetry.v2` configures stats filter for prometheus and stackdriver, so this is irrelevant for tracing. Actually, I should add this validation when I was adding Prometheus extension provider, but it's too late.
```suggestion
```